samples/gen-go/server/handlers.go

```diff
@@ -569,10 +604,18 @@ func statusCodeForHealthCheck(obj interface{}) int {
 
 	switch obj.(type) {
 
-	case models.DefaultBadRequest:
+	case *models.BadRequest:
 		return 400
-	case models.DefaultInternalError:
+
+	case *models.InternalError:
 		return 500
+
+	case models.BadRequest:
```

Comment:
note: added support for both the pointer and non-pointer form just to protect against the user doing something unexpected - since that wouldn't be caught at compile time